File: main/launcher/src/lib/commands.ts
import { invoke } from "@tauri-apps/api/core";
import type { Pm2Action, ProcessStatus } from "../types";

export async function getProcessStatus(): Promise<ProcessStatus> {
  return invoke<ProcessStatus>("get_process_status");
}

export async function pm2Command(action: Pm2Action): Promise<string> {
  return invoke<string>("pm2_command", { action });
}

File: main/launcher/src-tauri/src/commands.rs
use crate::pm2;

#[tauri::command]
pub async fn get_process_status() -> Result<pm2::ProcessStatus, String> {
    tokio::task::spawn_blocking(pm2::get_status)
        .await
        .map_err(|e| format!("Task join error: {}", e))?
}

#[tauri::command]
pub async fn pm2_command(action: String) -> Result<String, String> {
    let valid = ["start", "restart", "stop"];
    if !valid.contains(&action.as_str()) {
        return Err(format!("Invalid action: {}. Must be start, restart, or stop.", action));
    }

    tokio::task::spawn_blocking(move || pm2::pm2_action(&action))
        .await
        .map_err(|e| format!("Task join error: {}", e))?
}
